applications/FluidDynamicsApplication/custom_processes/embedded_skin_visualization_process.cpp

```diff
@@ -34,113 +35,295 @@ namespace Kratos
 {
 
 /* Public functions *******************************************************/
-EmbeddedSkinVisualizationProcess::EmbeddedSkinVisualizationProcess(
-    ModelPart& rModelPart,
-    ModelPart& rVisualizationModelPart,
-    const std::vector<Variable< double> >& rVisualizationScalarVariables,
-    const std::vector<Variable< array_1d<double, 3> > >& rVisualizationVectorVariables,
-    const std::vector<VariableComponent<VectorComponentAdaptor< array_1d< double, 3> > > >& rVisualizationComponentVariables,
-    const std::string& rShapeFunctions,
-    const bool ReformModelPartAtEachTimeStep) :
-    Process(),
-    mrModelPart(rModelPart),
-    mrVisualizationModelPart(rVisualizationModelPart),
-    mVisualizationScalarVariables(rVisualizationScalarVariables),
-    mVisualizationVectorVariables(rVisualizationVectorVariables),
-    mVisualizationComponentVariables(rVisualizationComponentVariables),
-    mShapeFunctions(rShapeFunctions),
-    mReformModelPartAtEachTimeStep(ReformModelPartAtEachTimeStep)
-{
-}
-
-EmbeddedSkinVisualizationProcess::EmbeddedSkinVisualizationProcess(
-    ModelPart& rModelPart,
-    ModelPart& rVisualizationModelPart,
-    Parameters& rParameters) :
-    Process(),
-    mrModelPart(rModelPart),
-    mrVisualizationModelPart(rVisualizationModelPart)
+Parameters EmbeddedSkinVisualizationProcess::GetDefaultSettings()
 {
-    Parameters default_parameters( R"(
+    Parameters default_settings(R"(
     {
-        "shape_functions"                     : "standard",
-        "reform_model_part_at_each_time_step" : false,
-        "visualization_variables"             : ["VELOCITY","PRESSURE","DISTANCE"]
+        "model_part_name"                       : "",
+        "visualization_model_part_name"         : "EmbeddedSkinVisualizationModelPart",
+        "reform_model_part_at_each_time_step"   : false,
+        "level_set_type"                        : "",
+        "shape_functions"                       : "",
+        "distance_variable_name"                : "",
+        "visualization_variables"               : ["VELOCITY","PRESSURE"],
+        "visualization_nonhistorical_variables" : []
     })");
 
-    rParameters.ValidateAndAssignDefaults(default_parameters);
+    return default_settings;
+}
 
-    // Validate if given shape functions value is admissible
-    std::set<std::string> available_shape_functions = {"standard","ausas"};
-    std::stringstream error_msg;
+ModelPart& EmbeddedSkinVisualizationProcess::CreateAndPrepareVisualizationModelPart(
+    Model& rModel,
+    const Parameters rParameters)
+{
+    // Set visualization model part data
+    const std::size_t buffer_size = 1;
+    std::string visualization_mp_name = rParameters["visualization_model_part_name"].GetString();
+    auto &r_origin_model_part = rModel.GetModelPart(rParameters["model_part_name"].GetString());
```

Comment:
```suggestion
    auto& r_origin_model_part = rModel.GetModelPart(rParameters["model_part_name"].GetString());
```
Isn't this the preferred style?

Reply:
Not by me (there is a declaration pathological case) :smile: . In any case I always try to stick to the preferred style. I will change it.